Apps/WebClient/src/ClientApp/src/components/timeline/linearTimeline.vue

```diff
@@ -120,10 +121,54 @@ export default class LinearTimelineComponent extends Vue {
             self.setPageFromDate(eventDate);
         });
 
+        this.eventBus.$on(
+            EventMessageName.TimelineCurrentDateUpdated,
+            function (currentDate: Date) {
+                self.goToPageByCurrentDate(currentDate);
+            }
+        );
+
         window.addEventListener("resize", this.handleResize);
         this.handleResize();
     }
 
+    private goToPageByCurrentDate(currentDate: Date) {
+        const selectedYearMonth = moment(currentDate).format("YYYY-MM");
+        let currentPage = 1;
+        let currentEntryIndex = 1;
+        let i = 0;
+        let foundLastEntryOfSelectedMonth = false;
+        const timelineEntriesLenght = this.timelineEntries.length;
+        // scan from the most recent entries to the older ones
+        while (i < timelineEntriesLenght && !foundLastEntryOfSelectedMonth) {
+            let entry = this.timelineEntries[i];
+            if (
+                entry.date !== undefined &&
+                entry.date.toString().indexOf(selectedYearMonth) == 0
+            ) {
+                console.log("found the last entry of the selected month");
+                foundLastEntryOfSelectedMonth = true;
+            } else {
+                currentEntryIndex++;
+                if (currentEntryIndex > this.numberOfEntriesPerPage) {
+                    currentEntryIndex = 1;
+                    currentPage++;
+                }
+            }
+            i++;
+        }
+        if (foundLastEntryOfSelectedMonth) {
+            // If found entry is at the bottom of the to-go page, move to the next page
+            if (
+                currentEntryIndex > this.numberOfEntriesPerPage - 3 &&
```

Comment:
we don't know if there are -3 entries. This could cause an index out of bounds

Reply:
do you think the "currentPage < this.numberOfPages" condition can back this up even if there are less than 3 items right?

Reply:
I remove this part completely then